core/src/main/java/com/dtolabs/rundeck/core/plugins/PluginRegistry.java

```diff
@@ -210,4 +210,12 @@ ValidatedPlugin validatePluginByName(String name, PluggableProviderService servi
      * @throws ProviderLoaderException
      */
     public PluginMetadata getPluginMetadata(String service, String provider) throws ProviderLoaderException;
+
+    /**
+     * Register a plugin into map using type and name as key to load it when requested
+     * @param type
+     * @param name
+     * @param beanName
+     */
+    void registryPlugin(String type, String name, String beanName);
```

Comment:
suggested rename of this method
```suggestion
    void registerPlugin(String type, String name, String beanName);
```